task5/src/main/java/com/clouway/adapter/http/TransactionHistory.java

```diff
@@ -0,0 +1,83 @@
+package com.clouway.adapter.http;
+
+import com.clouway.core.ConnectionProvider;
+import com.clouway.core.FundsRepository;
+import com.clouway.core.Session;
+import com.mysql.jdbc.jdbc2.optional.MysqlConnectionPoolDataSource;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+
+/**
+ * Created by Kristiyan Petkov  <kristiqn.l.petkov@gmail.com> on 13.06.16.
+ */
+@WebServlet(name = "TransactionHistory")
+public class TransactionHistory extends HttpServlet {
+  private FundsRepository fundsRepository;
+  private Integer offset;
+
+  public TransactionHistory(FundsRepository fundsRepository) {
+    this.fundsRepository = fundsRepository;
+  }
+
+  @Override
+  protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+    HttpSession session = request.getSession();
+
+    if (session.getAttribute("offset") == null) {
+      session.setAttribute("offset", 0);
+    }
+
+    offset = (Integer) session.getAttribute("offset");
```

Comment:
No need to use attributes. Use params like: 
`/operationHistoryPage?offset=0&limit=20` or
`/operationHistoryPage?page=2`
This way you keep history with the back button in the browser for example.